// Generated from beverage_service.ifex by covesa-ifex-core
// DO NOT EDIT - regenerate with: ./generate_proto.sh
syntax = "proto3";



package swdv.beverage_service;

// Namespace: beverage
// Beverage preparation control and monitoring

// Available beverage types
enum beverage_type_t {
  COFFEE = 0;
  ESPRESSO = 1;
  CAPPUCCINO = 2;
  TEA = 3;
  HOT_WATER = 4;
}

// Current state of beverage preparation
enum preparation_state_t {
  IDLE = 0;
  HEATING = 1;
  BREWING = 2;
  READY = 3;
  ERROR = 4;
}

// Beverage strength preference
enum strength_level_t {
  MILD = 0;
  REGULAR = 1;
  STRONG = 2;
  EXTRA_STRONG = 3;
}

// Configuration for beverage preparation
message beverage_config_t {
  beverage_type_t type = 1;
  strength_level_t strength = 2;
  uint32 temperature_celsius = 3;
  uint32 size_ml = 4;
}

// Current preparation status
message preparation_status_t {
  preparation_state_t state = 1;
  uint32 progress_percent = 2;
  uint32 remaining_seconds = 3;
  sint64 ready_at_timestamp = 4;
}

// Method: prepare_beverage
// Start beverage preparation
message prepare_beverage_request {
  beverage_config_t config = 1;
}

message prepare_beverage_response {
  bool accepted = 1;
  string preparation_id = 2;
  uint32 estimated_duration_seconds = 3;
}

service prepare_beverage_service {
  rpc prepare_beverage(prepare_beverage_request) returns (prepare_beverage_response);
}

// Method: get_preparation_status
// Get current preparation status
message get_preparation_status_request {
  string preparation_id = 1;
}

message get_preparation_status_response {
  preparation_status_t status = 1;
}

service get_preparation_status_service {
  rpc get_preparation_status(get_preparation_status_request) returns (get_preparation_status_response);
}

// Method: cancel_preparation
// Cancel ongoing preparation
message cancel_preparation_request {
  string preparation_id = 1;
}

message cancel_preparation_response {
  bool success = 1;
}

service cancel_preparation_service {
  rpc cancel_preparation(cancel_preparation_request) returns (cancel_preparation_response);
}

// Method: get_capabilities
// Get beverage system capabilities
message get_capabilities_request {
}

message get_capabilities_response {
  repeated beverage_type_t available_types = 1;
  uint32 water_level_percent = 2;
  bool requires_maintenance = 3;
}

service get_capabilities_service {
  rpc get_capabilities(get_capabilities_request) returns (get_capabilities_response);
}


